packages/salesforcedx-vscode-core/package.json

```diff
@@ -132,9 +132,14 @@
       ],
       "commandPalette": [
         {
-          "command": "sfdx.force.auth.web.login",
+          "command": "sfdx.force.auth.dev.hub",
           "when": "sfdx:project_opened"
         },
+        {
+          "command": "sfdx.force.auth.web.login",
+          "when":
+            "config.salesforcedx-vscode-core.change_set_based_tools.enabled"
```

Comment:
Is this a mistake? Looking at the diff it seems that sfdx.force.auth.web.login existed before but it had a different when clause. If we change it to "config.salesforcedx-vscode-core.change_set_based_tools.enabled" that means that it will only show when that pref is set. Is that intentional?

Reply:
NVM. I think I see what you are doing. You are actually changing the semantics of this command to do something different. That was what confused me. So previously, sfdx.force.auth.web.login was use to authorize a dev hub but now you are using it to actually authenticate against a prod/scratch org.

Two things:

1. The when clause should be config.salesforcedx-vscode-core.change_set_based_tools.enabled && sfdx:project_opened since you want this to be only invocable if both conditions are true. You still want an sfdx-project.json to be present.
2. In the future, we cannot repurpose command names. When the analytics from @lcampos goes in and he logs the commands, it will be very confusing to see the command names do different things. Since the analytics hasn't gone in yet, this should be OK.

Reply:
Sorry for the confusion about the renaming of the command name. That is a great point about analytics for the future. I have updated the conditions of the `when` clause to check that the user is in an sfdx project.